test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/MaxInputBufferLengthTests.cs

```diff
@@ -0,0 +1,233 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Security;
+using System.Net.Sockets;
+using System.Security.Authentication;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests
+{
+    public class MaxInputBufferLengthTests
+    {
+        private const int _dataLength = 10 * 1024 * 1024;
+
+        public static IEnumerable<object[]> LargeUploadData
+        {
+            get
```

Comment:
Please include some rationale for the test data you're creating here i.e. why those values and what you're trying to achieve with each one (or each group).

I'm fine with having the test like this (since we already follow this pattern in other places), but I'm not a fan of this approach. I prefer separate, descriptive tests (e.g. `UploadExpectsPause`, `BufferLengthControlWorksWithSsl`, etc.) that stand as a spec of what the code is supposed to do.